src/lib/pci_virtio_sock.c

```diff
@@ -1723,6 +1724,31 @@ static void apply_socket_config_set_buf_alloc(struct pci_vtsock_sock *s,
 	fprintf(stderr, "config: %s buf_alloc = %d\n", s->name, buf_alloc);
 }
 
+static void apply_socket_config_set_name(struct pci_vtsock_sock *s,
+					 struct vtsock_config_hdr *hdr)
+{
+	char *new_name;
+	char *msg_name = (char *)(hdr + 1);
+	size_t name_len = hdr->len - sizeof(struct vtsock_config_hdr);
+
+	name_len = strnlen(msg_name, name_len);
+
+	new_name = malloc(name_len + 22); /* max int string length + : + \0 */
```

Comment:
That would be nicer -- for some reason I forgot about `asprintf` temporarily! As discussed, I think we should probably just truncate long names and have then statically allocated in the sock struct.